indico/queries/teach_tasks.py

```diff
@@ -0,0 +1,227 @@
+                    rowIndex
+                    datafileId
+                    source
+                }
+            }
+        }
+    }
+    """
+
+    def __init__(self, questionaire_id: int, num_examples: int):
+        super().__init__(
+            query=self.query,
+            variables={
+                "questionaire_id": questionaire_id,
+                "num_examples": num_examples,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            examples = [
+                Example(**e)
+                for e in super().process_response(response)["questionnaires"][
+                    "questionnaires"
+                ][0]["examples"]
+            ]
+        except IndexError:
+            raise IndicoNotFound(
+                "Examples not found. Please check the ID you are using."
+            )
+        return examples
+
+
+class CreateQuestionaire(GraphQLRequest):
+    query = """
+        mutation(
+            $name: String!,
+            $dataset_id: Int!,
+            $questions: [QuestionInput]!,
+            $source_col_id: Int!,
+        ) {
+            createQuestionnaire (
+                datasetId: $dataset_id,
+                dataType: TEXT,
+                name: $name,
+                numLabelersRequired: 1,
+                questions: $questions,
+                sourceColumnId: $source_col_id,
+                instructions: ""
+            ) {
+                id
+            }
+        }
+    """
+
+    def __init__(self, name: str, dataset_id: int, source_column_id: int, targets: Any):
+        questions = [
+            {"type": "ANNOTATION", "targets": targets, "keywords": [], "text": name,}
+        ]
+        super().__init__(
+            query=self.query,
+            variables={
+                "name": name,
+                "dataset_id": dataset_id,
+                "questions": questions,
+                "source_col_id": source_column_id,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            questionnaire = Questionnaire(
+                **super().process_response(response)["createQuestionnaire"]
+            )
+        except IndexError:
+            raise IndicoNotFound("Failed to create Questionnaire")
+        return questionnaire
+
+
+class _GetDatasetInfo(GraphQLRequest):
+    query = """
+        query(
+            $id: Int!
+        ) {
+            dataset(id: $id) {
+                files {
+                    name
+                }
+            }
+        }
+
+    """
+
+    def __init__(self, dataset_id: int):
+        super().__init__(
+            query=self.query, variables={"id": dataset_id},
+        )
+
+
+class GetDatasetInfo(RequestChain):
+    previous = None
+
+    def __init__(self, id: int):
+        self.id = id
+        super().__init__()
+
+    def requests(self):
+        yield _GetDatasetInfo(self.id)
+
+
+class CreateDatasetTeach(RequestChain):
```

Comment:
We should separate creating a dataset from creating a teach task. We should have `CreateTeachTask` that takes as argument an existing dataset and not always require the request chain to create a dataset first... or at least make it optional whether to create a dataset in this chain